### ISSUE TITLE
Non existence column makes bug when using find/findOne

### ISSUE THREAD

**Issue by oghea** [typeorm/typeorm#2167]
**Issue type:**

[ ] question
[x ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
```
import { 
  Entity, Column, Index
} from "typeorm";

@Entity()
export class Device extends Entity {
  @Column({
    type: `varchar`,
    primary: true
  })
  id: string;
}
```

```
const device = new Device();
device.id = `12345`;
device.save();

Device.findOne({
  id: `12345`
}); // will return result from SELECT * FROM device where id = '12345' -- OK

// non existence column makes bug
Device.findOne({
  whatever: `12345`
}); // will return result from SELECT * FROM device -- BUG

Device.findOne({
  where: {
    id: `12345`
  }
}); // will return result from SELECT * FROM device where id = '12345' -- OK

// non existence column makes bug
Device.findOne({
  where: {
    whatever: `12345`
  }
}); // will return result from SELECT * FROM device -- BUG

```

**Comment by pleerock** [comment]
```ts
// non existence column makes bug
Device.findOne({
  whatever: `12345`
});
```

non existing column shall give you a compiler error.

**Comment by pleerock** [comment]
@terrybleger don't overstate please. If there is a missing error wrapping or check for column existence it does not mean this library should not be used where it should be.

Btw in 0.3.0 we expect completely reworked find options, and it will give a runtime error if some properties or columns are missing.

**Comment by bjuffe** [comment]
@pleerock / @Kononnable just to clarify, this is a problem that will be fixed in 0.3.0? Tried to search for an open issue with this but couldn't find it and having a hard time understanding why its closed.

**Comment by Kononnable** [comment]
Yes, this should be fixed in 0.3.0. Find options were changed soon after splitting the branches so it should be already available in `@next`.

**Comment by bjuffe** [comment]
@Kononnable works like a charm when testing out next, thank you for your swift reply!
